src/main/java/io/bit3/jsass/adapter/NativeAdapter.java

```diff
@@ -61,7 +62,7 @@ public Output compile(StringContext context, ImportStack importStack)
    * @return The given {@link Output}
    * @throws CompilationException If an error was found in the given {@link Output}
    */
-  private Output checkForError(Output output) throws CompilationException {
+  private SuccessOutput checkForError(Output output) throws CompilationException {
```

Comment:
This method is the logical key of this PR.
It takes the unclassified Output and either returns SuccessOutput, if the compilation was successful, or it throws an exception